trantor/net/inner/TcpConnectionImpl.cc

```diff
@@ -1503,12 +1597,118 @@ void TcpConnectionImpl::sendFileInLoop(const BufferNodePtr &filePtr)
         return;
     }
 #endif
-#ifndef _WIN32
-    lseek(filePtr->sendFd_, filePtr->offset_, SEEK_SET);
+    // Send stream
+    if (filePtr->streamCallback_)
+    {
+        if (!fileBufferPtr_)
+        {
+            fileBufferPtr_ = std::make_unique<std::vector<char>>();
+            fileBufferPtr_->reserve(16 * 1024);
+        }
+        char pszChunkSize[8];
+        while ((filePtr->fileBytesToSend_ > 0) || !fileBufferPtr_->empty())
+        {
+            // get next chunk
+            if (fileBufferPtr_->empty())
+            {
+                fileBufferPtr_->resize(16 * 1024);
+                std::size_t nData;
+                if (filePtr->chunked_)  // Put chunk size + chunk data in buffer
+                {
+                    nData = filePtr->streamCallback_(
+                        fileBufferPtr_->data() +
+                            6,  // reserve 6 for header (0-4000 hex size + \r\n)
+                        fileBufferPtr_->size() -
+                            8);  // reserve header + 2 for trailer (\r\n)
+                    fileBufferPtr_->resize(nData + 6);
+                    fileBufferPtr_->push_back('\r');
+                    fileBufferPtr_->push_back('\n');
```

Comment:
The underlying functions should not know the details of the upper-layer protocol, I prefer to implement these details in drogon (and the trunked flag is not needed) 
@marty1885 What do you think?

Reply:
Agree. TcpConnection should have no knowledge of HTTP.